server/sonar-server/src/main/java/org/sonar/ce/configuration/WorkerCountProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.ce.configuration;
+
+public interface WorkerCountProvider {
+  int get();
```

Comment:
In fact there were a misundersanding : on my box the commit https://github.com/SonarSource/sonarqube/pull/2217/commits/48631ffb5004218571fddf871e491176f8fd27db was not existing (probably pull at the wrong time) , so `CeConfigurationImpl` was still reading number of workers from internal properties instead of using `WorkerCountProvider#get()`.

But indeed, everything is ok !